<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"[]>
<html xmlns="http://www.w3.org/1999/xhtml" dir="ltr" lang="en-US" xml:lang="en">
<head>
    <!--
    Created by Artisteer v3.0.0.39952
    Base template (without user's data) checked by http://validator.w3.org : "This page is valid XHTML 1.0 Transitional"
    -->
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
    <title>Discussion</title>
    <meta name="description" content="Description" />
    <meta name="keywords" content="Keywords" />


    <link rel="stylesheet" href="css-javascripts/style.css" type="text/css" media="screen" />
    <!--[if IE 6]><link rel="stylesheet" href="style.ie6.css" type="text/css" media="screen" /><![endif]-->
    <!--[if IE 7]><link rel="stylesheet" href="style.ie7.css" type="text/css" media="screen" /><![endif]-->

    <script type="text/javascript" src="css-javascripts/jquery.js"></script>
    <script type="text/javascript" src="css-javascripts/script.js"></script>
</head>
<body>
<div id="art-page-background-glare">
    <div id="art-page-background-glare-image"> </div>
</div>
<div id="art-main">
    <div class="art-sheet">
        <div class="art-sheet-tl"></div>
        <div class="art-sheet-tr"></div>
        <div class="art-sheet-bl"></div>
        <div class="art-sheet-br"></div>
        <div class="art-sheet-tc"></div>
        <div class="art-sheet-bc"></div>
        <div class="art-sheet-cl"></div>
        <div class="art-sheet-cr"></div>
        <div class="art-sheet-cc"></div>
        <div class="art-sheet-body">
            <div class="art-header">
                <div class="art-header-clip">
                <div class="art-header-center">
                    <div class="art-header-png"></div>
                    <div class="art-header-jpeg"></div>
                </div>
                </div>
                <div class="art-logo">
                                 <h1 class="art-logo-name"><a href="../main_page.html"><font color="black"><center>Evolutionary Systems Biology of Bacterial Metabolism</center></font></a></h1>
                                                 <h2 class="art-logo-text"><center><font color="black">Ali Ahmad Faruqi</font></center></h2>
												 <h2 class="art-logo-text"><center><font color="black">Data Analysis Project</font></center></h2>
												 <h2 class="art-logo-text"><center><font color="black">MSc Bioinformatics and Theoretical Systems Biology 2011-12</font></center></h2>
                                </div>
            </div>
            <div class="cleared reset-box"></div>
<div class="art-nav">
	<div class="art-nav-l"></div>
	<div class="art-nav-r"></div>
<div class="art-nav-outer">
	<ul class="art-hmenu">
		<li>
			<a href="../main_page.html"><span class="l"></span><span class="r"></span><span class="t">Introduction</span></a>
		</li>	
		<li>
			<a href="./results.html"><span class="l"></span><span class="r"></span><span class="t">Results</span></a>
		</li>
		<li>
			<a href="./discussion.html" class="active"><span class="l"></span><span class="r"></span><span class="t">Discussion</span></a>
		</li>
		<li>
			<a href="./materials-methods.html"><span class="l"></span><span class="r"></span><span class="t">Material and Methods</span></a>
		</li>
		<li>
			<a href="./references.html"><span class="l"></span><span class="r"></span><span class="t">References</span></a>
		</li>
		<li>
			<a href="./acknowledgement.html"><span class="l"></span><span class="r"></span><span class="t">Acknowledgement</span></a>
		</li>
	</ul>
</div>
</div>
<div class="cleared reset-box"></div>
<div class="art-content-layout">
                <div class="art-content-layout-row">
                    <div class="art-layout-cell art-sidebar1">
<div class="art-vmenublock">
    <div class="art-vmenublock-body">
                <div class="art-vmenublockheader">
                    <h3 class="t">Sections</h3>
                </div>
                <div class="art-vmenublockcontent">
                    <div class="art-vmenublockcontent-body">
                <ul class="art-vmenu">
	<li>
		<a href="../main_page.html"><span class="l"></span><span class="r"></span><span class="t">Introduction</span></a>
	</li>	
	<li>
		<a href="./results.html"><span class="l"></span><span class="r"></span><span class="t">Results</span></a>
	</li>
	<li>
		<a href="./discussion.html" class="active"><span class="l"></span><span class="r"></span><span class="t">Discussion</span></a>
	</li>
	<li> 
		<a href="./materials-methods.html"><span class="l"></span><span class="r"></span><span class="t">Material and Methods</span></a>
	</li>
	<li>
		<a href="./references.html"><span class="l"></span><span class="r"></span><span class="t">References</span></a>
	</li>
	<li>
		<a href="./acknowledgement.html"><span class="l"></span><span class="r"></span><span class="t">Acknowledgement</span></a>
	</li>
</ul>
                
                                		<div class="cleared"></div>
                    </div>
                </div>
		<div class="cleared"></div>
    </div>
</div>
<div class="art-block">
    <div class="art-block-body">
                <div class="art-blockheader">
                    <h3 class="t">Keywords</h3>
                </div>
                <div class="art-blockcontent">
                    <div class="art-blockcontent-body">
				<p><b>Ancestral State Reconstruction</b></p>
				<p><b>Bacterial Adaptation</b></p>
				<p><b>Metabolic Traits Data</b></p>
                                		<div class="cleared"></div>
                    </div>
                </div>
		<div class="cleared"></div>
    </div>
</div>



                      <div class="cleared"></div>
                    </div>
                    <div class="art-layout-cell art-content">
<div class="art-post">
    <div class="art-post-body">
<div class="art-post-inner art-article">
                                <h2 class="art-postheader">
                Discussion
                                </h2>
                <div class="cleared"></div>
                                <div class="art-postcontent">
<p align="justify"><b>1) Distribution of reactions in organisms based on lifestyle</b></p>
<p align="justify">The Mann Whitney U Test performed to compare different habitats and respiration types of bacteria, provided interesting results. A significant statistical difference was observed in 4 group comparisons related to bacterial habitat and 6 group comparisons related to bacterial mode of respiration. </p>
<p align="justify">When comparing the modes of respirations, a highly significant statistical difference was observed between the distribution of reactions belonging to facultative anaerobes and anaerobes. This implied that the number of reactions present in facultative anaerobic organisms is greater than the number of reactions present in anaerobic organisms than would be expected by chance alone. This is also evident from the boxplot figures as the distribution of reactions in facultative anaerobes is greater than anaerobes. It also makes sense from the biological perspective as facultative anaerobes do possess the metabolic reactions to synthesize ATP in presence of oxygen whereas these reactions are absent in anaerobes [7]. </p>
<p align="justify">Likewise, in the comparison of bacterial habitats, a highly significant statistical difference was observed between the distribution of reactions belonging to organisms living in multiple habitats and host-associated organisms. The boxplot of host-associated organisms shows the greatest variation as compared to other groups. One possible explanation, which is dealt in greater detail in the following section, is that host-associated bacteria can be endocellular parasites having very few metabolic reactions as well as multi-drug resistant, toxin secreting pathogens possessing a great number of metabolic reactions. Therefore, the distribution of reactions present in host-associated organisms is quite spread out as compared to other groups resulting in statistical significance of results. </p>
<p align="justify"><b>2) Ancestral state reconstruction analysis</b></p>
<p align="justify">The results obtained from ancestral state reconstruction analysis of metabolism of different bacteria appear to tally with our biological knowledge about different bacteria. Using the aforementioned threshold score, the top five branches that showed the greatest number of gain and loss events were terminal branches leading to various extant bacteria. </p>
<p align="justify">The greatest gain is observed in the terminal branch leading to the bacterium, <i>Clostridium difficile</i> CD196. This change could be related to its antibiotic resistance and ability to produce different types of toxins and compounds such p-cresol [8] in comparison with other members of its genus such as <i>Clostridium beijerinckii</i> NCIMB 8052, which does possess the metabolic reactions to create these toxins. Significant gains were also seen <i>Pseudomonas aeruginosa</i> PAO1, which is a metabolically versatile organism capable of breaking down a wide range of organic molecules including benzoate [9]. It is also an opportunistic pathogen, being resistant to antibiotics.</p>
<p align="justify">Last but not the least, <i>Salmonella enterica subsp. enterica serovar Typhimurium str. LT2</i> also showed significant gain in metabolic reactions. As a pathogen, it is a leading cause of human gastroenteritis producing secreted proteins that cause pathogenesis [10].</p>
<p align="justify">It is interesting to note that organisms that are protein secreting pathogens have a significant gain in metabolic reactions during the course of evolution. Evolutionary studies about the metabolism of toxin production can shed more light on this. </p>
<p align="justify">In terms of loss, the greatest loss is observed in <i>Onion yellows phytoplasma</i> OY−M, which is plant pathogen, has minimal gene set and its “<i>genome encodes even fewer metabolic functions than mycoplasma genomes</i>” [11].</p>
<p align="justify">The second greatest loss is observed in <i>Tropheryma whipplei str. Twist</i>, a human pathogen with reduced genome. This bacterium has deficiency in the metabolism of amino acids and lacks homologs for thioredoxin and thioredoxin reductase enzyme [12]. </p>
<p align="justify">Significant loss is also observed in <i>Bartonella bacilliformis</i> KC583, which is a pathogen causing diseases such as Carrion's disease and Oroya fever [13] and in <i>Bifidobacterium longum</i> NCC2705, which can be explained by its adapted life-style, as it is lives in the gastrointestinal tract of mammals [14]. </p>
<p align="justify">Losses were also observed in <i>Nitrobacter winogradskyi</i> Nb−255, which is consistent with biological data as this bacterium is lacks the Embden-Meyerhof and Entner-Doudoroff metabolic pathways and cannot actively uptake sugars [15]. </p>
<p align="justify">Furthermore, losses were also observed in <i>Buchnera aphidicola str. APS</i> (Acyrthosiphon pisum), which is an endocellular bacterial symbiont of aphids [16] and in <i>Wigglesworthia glossinidia</i>, which is an endocellular obligate symbiont of the tsetse flies [17]. </p>
<p align="justify">Overall, as seen from the major losses, organisms that have small genome size and have adapted to live inside hosts have lost metabolic reactions over the course of evolution, as they meet their survival requirements from hosts they dwell in. </p>
<p align="justify"><b>3) Metabolic Traits Phylogeny</b></p>
<p align="justify">Based on 23S rRNA sequence similarity, many organisms appear closely related to each other on the phylogenetic tree. However, the metabolism data seems to suggest significant departure of many organisms with each other. As a consequence of those departures, majority of gains and losses are summarized on the terminal tree branches, indicating speciation events, in terms of metabolism. This naturally leads us to compare a phylogeny based on 23S rRNA sequences with a phylogeny based on metabolism data, to see how evolutionary distant or related different organisms are. </p>
<p align="justify">The results obtained clearly show that even though two organisms may appear closely related based on 23S rRNA sequence similarity, they are evolutionary distant in terms of their metabolic capabilities. A clear example is between <i>Mycoplasma pulmonis</i> UAB CTIP and <i>Onion yellows phytoplasma</i> OY-M. They appear evolutionary distant on the 23S rRNA phylogeny but are very closely related on the metabolic traits phylogeny as they both have reduced genome size and similar metabolic profiles. Similarly, the entire clade of <i>Staphylococcus aureus</i>, which shows reasonable amount of gains and losses at the last common ancestral node compared to other clades, appears metabolically distinct and evolutionary distant from other organisms on the metabolic traits tree. </p>
<p align="justify"><b>4) Conclusion</b></p>
<p align="justify">The ancestral state reconstruction results and metabolic traits phylogeny have been able to explain the significant gains and losses in various organisms during their evolutionary history. The findings have correlated well with the biological knowledge of the lifestyles of these organisms.  </p>
<p align="justify">The adaptation of bacteria to different conditions has led to a considerable gain and/or loss of reactions over time. Considerable gain has been observed in bacteria that are drug-resistant and toxin-producing whereas considerable loss has been observed in bacteria that persist as endosymbionts. The metabolic traits based phylogeny has provided insight into examples of convergent evolution with respect to bacterial metabolism. Future work can focus on integrating more datasets pertaining to the transcriptome and proteome of the bacteria, to generate a system level phylogeny and explore the evolutionary relatedness of different organisms. </p>
                </div>
                <div class="cleared"></div>
                </div>

		<div class="cleared"></div>
    </div>
</div>

                      <div class="cleared"></div>
                    </div>
                </div>
            </div>
            <div class="cleared"></div>
            <div class="art-footer">
                <div class="art-footer-t"></div>
                <div class="art-footer-l"></div>
                <div class="art-footer-b"></div>
                <div class="art-footer-r"></div>
                <div class="art-footer-body">
                    <a href="#" class="art-rss-tag-icon" title="RSS"></a>
                            <div class="art-footer-text">
                                <p>Website Template created with Artisteer.</p>http://www.2createawebsite.com/artisteer<p></p><p>Copyright © 2011. All Rights Reserved.</p>
                                                            </div>
                    <div class="cleared"></div>
                </div>
            </div>
    		<div class="cleared"></div>
        </div>
    </div>
    <div class="cleared"></div>
</div>

</body>
</html>
