<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"[]>
<html xmlns="http://www.w3.org/1999/xhtml" dir="ltr" lang="en-US" xml:lang="en">
<head>
    <!--
    Created by Artisteer v3.0.0.39952
    Base template (without user's data) checked by http://validator.w3.org : "This page is valid XHTML 1.0 Transitional"
    -->
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
    <title>Material and Methods</title>
    <meta name="description" content="Description" />
    <meta name="keywords" content="Keywords" />


    <link rel="stylesheet" href="css-javascripts/style.css" type="text/css" media="screen" />
    <!--[if IE 6]><link rel="stylesheet" href="style.ie6.css" type="text/css" media="screen" /><![endif]-->
    <!--[if IE 7]><link rel="stylesheet" href="style.ie7.css" type="text/css" media="screen" /><![endif]-->

    <script type="text/javascript" src="css-javascripts/jquery.js"></script>
    <script type="text/javascript" src="css-javascripts/script.js"></script>
</head>
<body>
<div id="art-page-background-glare">
    <div id="art-page-background-glare-image"> </div>
</div>
<div id="art-main">
    <div class="art-sheet">
        <div class="art-sheet-tl"></div>
        <div class="art-sheet-tr"></div>
        <div class="art-sheet-bl"></div>
        <div class="art-sheet-br"></div>
        <div class="art-sheet-tc"></div>
        <div class="art-sheet-bc"></div>
        <div class="art-sheet-cl"></div>
        <div class="art-sheet-cr"></div>
        <div class="art-sheet-cc"></div>
        <div class="art-sheet-body">
            <div class="art-header">
                <div class="art-header-clip">
                <div class="art-header-center">
                    <div class="art-header-png"></div>
                    <div class="art-header-jpeg"></div>
                </div>
                </div>
                <div class="art-logo">
                                 <h1 class="art-logo-name"><a href="../main_page.html"><font color="black"><center>Evolutionary Systems Biology of Bacterial Metabolism</center></font></a></h1>
                                                 <h2 class="art-logo-text"><center><font color="black">Ali Ahmad Faruqi</font></center></h2>
												 <h2 class="art-logo-text"><center><font color="black">Data Analysis Project</font></center></h2>
												 <h2 class="art-logo-text"><center><font color="black">MSc Bioinformatics and Theoretical Systems Biology 2011-12</font></center></h2>
                                </div>
            </div>
            <div class="cleared reset-box"></div>
<div class="art-nav">
	<div class="art-nav-l"></div>
	<div class="art-nav-r"></div>
<div class="art-nav-outer">
	<ul class="art-hmenu">
		<li>
			<a href="../main_page.html"><span class="l"></span><span class="r"></span><span class="t">Introduction</span></a>
		</li>	
		<li>
			<a href="./results.html"><span class="l"></span><span class="r"></span><span class="t">Results</span></a>
		</li>
		<li>
			<a href="./discussion.html"><span class="l"></span><span class="r"></span><span class="t">Discussion</span></a>
		</li>
		<li>
			<a href="./materials-methods.html" class="active"><span class="l"></span><span class="r"></span><span class="t">Material and Methods</span></a>
		</li>
		<li>
			<a href="./references.html"><span class="l"></span><span class="r"></span><span class="t">References</span></a>
		</li>
		<li>
			<a href="./acknowledgement.html"><span class="l"></span><span class="r"></span><span class="t">Acknowledgement</span></a>
		</li>
	</ul>
</div>
</div>
<div class="cleared reset-box"></div>
<div class="art-content-layout">
                <div class="art-content-layout-row">
                    <div class="art-layout-cell art-sidebar1">
<div class="art-vmenublock">
    <div class="art-vmenublock-body">
                <div class="art-vmenublockheader">
                    <h3 class="t">Sections</h3>
                </div>
                <div class="art-vmenublockcontent">
                    <div class="art-vmenublockcontent-body">
                <ul class="art-vmenu">
	<li>
		<a href="../main_page.html"><span class="l"></span><span class="r"></span><span class="t">Introduction</span></a>
	</li>	
	<li>
		<a href="./results.html"><span class="l"></span><span class="r"></span><span class="t">Results</span></a>
	</li>
	<li>
		<a href="./discussion.html"><span class="l"></span><span class="r"></span><span class="t">Discussion</span></a>
	</li>
	<li> 
		<a href="./materials-methods.html" class="active"><span class="l"></span><span class="r"></span><span class="t">Material and Methods</span></a>
	</li>
	<li>
		<a href="./references.html"><span class="l"></span><span class="r"></span><span class="t">References</span></a>
	</li>
	<li>
		<a href="./acknowledgement.html"><span class="l"></span><span class="r"></span><span class="t">Acknowledgement</span></a>
	</li>
</ul>
                
                                		<div class="cleared"></div>
                    </div>
                </div>
		<div class="cleared"></div>
    </div>
</div>
<div class="art-block">
    <div class="art-block-body">
                <div class="art-blockheader">
                    <h3 class="t">Keywords</h3>
                </div>
                <div class="art-blockcontent">
                    <div class="art-blockcontent-body">
                <p><b>Ancestral State Reconstruction</b></p>
				<p><b>Mesquite</b></p>
				<p><b>Python</b></p>
				<p><b>Phylogeny</b></p>
				<p><b>R</b></p>
                                		<div class="cleared"></div>
                    </div>
                </div>
		<div class="cleared"></div>
    </div>
</div>



                      <div class="cleared"></div>
                    </div>
                    <div class="art-layout-cell art-content">
<div class="art-post">
    <div class="art-post-body">
<div class="art-post-inner art-article">
                                <h2 class="art-postheader">
                Material and Methods
                                </h2>
                <div class="cleared"></div>
                                <div class="art-postcontent">
<p align="justify"> A number of steps were followed to extract, analyze and summarize the data. The following sub-sections describe the procedure in detail.</p>
<p align="justify"><b> 1) Data Extraction from Database</b></p>
<p align="justify"> Tables containing information about reactions were downloaded from Mind The Gap Database using the relevant MySQL queries [18]. Scripts were written in Python [19] to extract and organize the name and number of species and reactions present within the database tables. </p>
<p align="justify"><b>2) Sequence Retrieval and Organization</b></p>
<p align="justify">23S rRNA sequences for all the 141 organisms present in Mind The Gap Database along with the out-group organism (Thermoplasma acidophilum) were obtained from the NCBI Nucleotide Database. The sequences were then written to a single, FASTA format output file. The sequences were retrieved from NCBI using the Biopython module [20] of Python incorporated in Python script that produced the FASTA file. </p>
<p align="justify"><b>3) Multiple Sequence Alignment</b></p>
<p align="justify">Multiple sequence alignment of the 23S rRNA sequences of 141 organisms plus one out-group organism was obtained through MAFFT 6.611 [21]. A threshold score of E=8.4e-11 was used.</p>
<p align="justify"><b>4) Phylogeny Construction</b></p>
<p align="justify">Based on the results of multiple sequence alignment, a phylogeny was constructed using PhyML 2.4.4 [22]. PhyML uses a maximum likelihood approach to construct phylogenies. Bootstrappping was performed a 100 times on the tree to obtain the most likely phylogeny. For visualization of the phylogeny obtained and creation of phylogeny images, Dendroscope was used [23]. </p>
<p align="justify"><b>5) Ancestral State Reconstruction</b></p>
<p align="justify">For ancestral state reconstruction, Mesquite system was used [24]. Mesquite is a powerful, modular software to perform various analyses in evolutionary biology such as phylogenetic analyses and population genetics analyses. A character matrix was created for all the 2556 metabolic reactions using Python scripts. If a reaction was present in an organism, the number one was stored and if it a reaction was absent, a zero was stored.</p>
<p align="justify">Maximum Likelihood Ancestral State Reconstruction was performed for the categorical, discrete dataset of reactions. Likelihood reconstruction methods look for ancestor states that maximize the probability of producing the current state, under a defined model of evolution [25] [4]. </p>
<p align="justify">Mesquite supports two models of evolution: the Markov k-state 1 parameter model (Mk1 model) and the Asymmetrical Markov k-state 2 parameter model (AsymmMK model). The AsymmMK model has two parameters. One parameter is for the rate of increase in state (from 0 to 1) and the other parameter is for decrease in state (from 1 to 0). For the given dataset, the AsymmMK model was, hence, used as it allowed to investigate the bias in gains versus loses [24]. </p>
<p align="justify">In the ancestral state reconstruction, the out-group was removed from the phylogeny and the reaction traits for the out-group were not specified in the character matrix.</p>
<p align="justify"><b>6) Metabolic Traits Phylogeny Construction</b></p>
<p align="justify">The construction of phylogeny based on metabolic traits was done through Pars programme in PHYLIP package [26]. Pars adopts a parsimony approach to construct phylogenetic trees, based on the Wagner parsimony method with multiple states. The Pars programme produced a total of 12 trees, from which a consensus tree was obtained using CONSENSE programme in PHYLIP package [26].</p>
<p align="justify"><b>7) Analysis and Summarization of Results</b></p>
<p align="justify">All figures were created in R [27]. Additional data about the habitat and respiration types of organisms included in the study were provided by Dr. Pinney. δP values were calculated using scripts written in Python. The coloured phylogenies indicating the gain and loss of reactions based on δP values were made using the ape package in R. [27]</p>
<p align="justify"><b>8) Webpage Development</b></p>
<p align="justify">The template of webpage was obtained from the Artisteer [28] website and modified accordingly.</p>
                </div>
                <div class="cleared"></div>
                </div>

		<div class="cleared"></div>
    </div>
</div>

                      <div class="cleared"></div>
                    </div>
                </div>
            </div>
            <div class="cleared"></div>
            <div class="art-footer">
                <div class="art-footer-t"></div>
                <div class="art-footer-l"></div>
                <div class="art-footer-b"></div>
                <div class="art-footer-r"></div>
                <div class="art-footer-body">
                    <a href="#" class="art-rss-tag-icon" title="RSS"></a>
                            <div class="art-footer-text">
                                <p>Website Template created with Artisteer.</p>http://www.2createawebsite.com/artisteer<p></p><p>Copyright © 2011. All Rights Reserved.</p>
                                                            </div>
                    <div class="cleared"></div>
                </div>
            </div>
    		<div class="cleared"></div>
        </div>
    </div>
    <div class="cleared"></div>
</div>

</body>
</html>
